pennylane/_qubit_device.py

```diff
@@ -960,3 +975,62 @@ def adjoint_jacobian(self, tape, starting_state=None, use_device_state=False):
             op.inv()
 
         return jac
+
+    def custom_process(
+        self,
+        post_processing_func,
+        base_measurement,
+        observable=None,
+        wires=None,
+        shot_range=None,
+        bin_size=None,
+    ):
+
+        if base_measurement == "sample":
+            measurement_obj = qml.sample(op=observable, wires=wires)
+            res = self.sample(measurement_obj, shot_range=shot_range, bin_size=bin_size)
+
+        elif base_measurement == "expval":
+            raise NotImplementedError(
+                f"custom_process currently does not support measuring {base_measurement}"
+            )
+        #     res = self.expval(observable, shot_range=shot_range, bin_size=bin_size)
+
+        elif base_measurement == "var":
+            raise NotImplementedError(
+                f"custom_process currently does not support measuring {base_measurement}"
+            )
+        #     res = self.var(observable, shot_range=shot_range, bin_size=bin_size)
+
+        elif base_measurement == "prob":
+            raise NotImplementedError(
+                f"custom_process currently does not support measuring {base_measurement}"
+            )
+        #     res = self.probability(wires=observable.wires, shot_range=shot_range, bin_size=bin_size)
+
+        elif base_measurement == "state":
+            raise NotImplementedError(
+                f"custom_process currently does not support measuring {base_measurement}"
+            )
+        #     if self.wires.labels != tuple(range(self.num_wires)):
+        #         raise qml.QuantumFunctionError(
+        #             "Returning the state is not supported when using custom wire labels"
+        #         )
+        #     res = self.access_state(wires=observable.wires)
+
+        else:
+            raise ValueError(
+                f"base_measurement should be one of [expval, var, sample, prob, state], "
+                f"got: {base_measurement}"
+            )
+
+        try:
+            post_processed_res = post_processing_func(res)
+        except BaseException as e:
+            print(
+                "Encountered the following exception when applying the post-processing function"
+                f" to the measured {base_measurement}:\n {e} \n"
+            )
+            return None
+
+        return post_processed_res
```

Comment:
Any feedback on this? I kept it in to make sure that any errors that get thrown by the user defined `post_processing_func` can be captured and we can produce a more descriptive error message.